src/main/java/edu/cornell/kfs/tax/batch/dataaccess/impl/TransactionDetailProcessorDaoJdbcImpl.java

```diff
@@ -0,0 +1,309 @@
+                .join(VendorHeader.class,
+                        Criteria.equal(VendorField.vendorHeaderGeneratedIdentifier_forDetail,
+                                VendorField.vendorHeaderGeneratedIdentifier_forHeader))
+                .where(
+                        Criteria.equal(VendorField.vendorHeaderGeneratedIdentifier_forDetail, Types.INTEGER,
+                                vendorId.getLeft()),
+                        Criteria.or(
+                                Criteria.equal(VendorField.vendorDetailAssignedIdentifier, Types.INTEGER,
+                                        vendorId.getRight()),
+                                Criteria.equal(VendorField.vendorParentIndicator, KRADConstants.YES_INDICATOR_VALUE)
+                        )
+                )
+                .orderBy(QuerySort.ascendingNullsFirst(VendorField.vendorParentIndicator))
+                .build();
+    }
+
+    private VendorQueryResults extractVendorQueryResults(final ResultSet resultSet,
+            final Pair<Integer, Integer> vendorId,
+            final TaxDtoDbMetadata vendorMetadata) throws SQLException {
+        final VendorDetailLiteMapper vendorMapper = new VendorDetailLiteMapper(
+                encryptionService, vendorMetadata, resultSet);
+        if (!vendorMapper.moveToNextRow()) {
+            return new VendorQueryResults(null, null);
+        }
+
+        final VendorDetailLite vendorDetail = vendorMapper.readCurrentRow();
+        final VendorDetailLite parentDetail;
+        if (!vendorMapper.moveToNextRow()) {
+            if (vendorId.getRight().equals(vendorDetail.getVendorDetailAssignedIdentifier())) {
+                parentDetail = null;
+                return new VendorQueryResults(vendorDetail, null);
+            } else {
+                parentDetail = vendorDetail;
+                return new VendorQueryResults(null, parentDetail);
+            }
+        } else {
+            parentDetail = vendorMapper.readCurrentRow();
+            if (vendorMapper.moveToNextRow()) {
+                LOG.warn("extractVendorDetailResults, More than 2 vendors were found by a query that should have "
+                        + "only fetched vendor {}-{} and/or its parent. The remaining results will be ignored.",
+                        vendorId.getLeft(), vendorId.getRight());
+            }
+
+            if (!parentDetail.isVendorParentIndicator()) {
+                LOG.warn("extractVendorDetailResults, Vendor {}-{} was expected to be the parent of vendor {}-{} "
+                        + "but it wasn't. It will be ignored.",
+                        parentDetail.getVendorHeaderGeneratedIdentifier(),
+                        parentDetail.getVendorDetailAssignedIdentifier(), vendorId.getLeft(), vendorId.getRight());
+                return new VendorQueryResults(vendorDetail, null);
+            } else {
+                return new VendorQueryResults(vendorDetail, parentDetail);
+            }
+        }
+    }
+
+
+
```

Comment:
I am okay with the multiple newlines if that was your intention? There is other places in this file where this situation occurs.

Reply:
Yes, I intended to have 3-line newline sections in various areas to help with some of the code grouping.